HW06.Rmd

```diff
@@ -1,92 +1,152 @@
+
+```{r map gene.ids to gene.symbols using ensembldb, message = FALSE}
+
+#install.packages("BiocManager")
+#BiocManager::install("ensembldb")
+#BiocManager::install("EnsDb.Hsapiens.v75")
+library(ensembldb)
+library(EnsDb.Hsapiens.v75)
+
+ens.GBM.transcripts <- GBM.transcripts %>% 
+  mutate(gene_id = gsub(pattern = "\\..*", "", .$gene_id))
+
+map <- ensembldb::select(EnsDb.Hsapiens.v75, keys = ens.GBM.transcripts$gene_id,
+                         keytype = "TXID", columns = c("SYMBOL", "TXID"))
+
+ens.mapped_GBM <- left_join(ens.GBM.transcripts, map, by = c("gene_id" = "TXID")) %>% 
+  dplyr::select(-1) %>% 
+  dplyr::select(gene_symbol = SYMBOL, everything())
+
+ens.mapped_GBM <- ens.mapped_GBM[!duplicated(ens.mapped_GBM$gene_symbol),] #remove duplicated gene symbols
+  #these are removed instead of averaged because they simply do not correlate particularly well. 
+ens.mapped_GBM <- ens.mapped_GBM[!is.na(ens.mapped_GBM$gene_symbol),] #remove NA values
+
+```
+
+
+
+Do the t-test and make a table of the t-test results!
+
+```{r t.test for difference between a bunch of genes}
+#run this code to unload the libraries from before, it might be helpful because the select() function from dplyr might be hidden otherwise
+detach(package:EnsDb.Hsapiens.v75, unload = T)
+detach(package:ensembldb, unload = T)
+
+#add in your own gene of interest!!! 
+genes_of_interest <- c("OR4G4P", "IRX1", "OSR1", "DCHS2", "BRINP3", "TOB2P1", "FOXD1", "ZFPM2", "GLB1", "ALG5", "TRIM4", "ADARB2", "PCDHGA11", "IDH1", "EGFR", "MGMT", "TERT", "PTEN", "TP53", "RB1", "ATRX", "PDGFRA", "PIK3CA", "MICA", "CDKN2A", "EZH2", "BRD2")
+
+# I found there are 2 "PTEN", so I deleted one. Hope that is okay.
+
+GBM.genes.of.interest <- filter(ens.mapped_GBM, gene_symbol %in% genes_of_interest)
+
+#Now perform a t-test between the H3K mutated and the wt samples. There are many ways to do this actually, you can use a for loop or you could do the tidy alternative with broom(), but the for loop is probably the easiest
+
+H3K27M <- GBM.genes.of.interest %>% 
+  dplyr::select(contains(".H3K27M."), gene_symbol) %>% 
+  gather(key = "Sample", value = "Expression27", -gene_symbol) %>% 
+  mutate(type = "H3K27M")
+  
+WT <- GBM.genes.of.interest %>% 
+    dplyr::select(contains(".WT."), gene_symbol) %>% 
+  gather(key = "Sample", value = "Expression27", -gene_symbol) %>% 
+  mutate(type = "WT")
+
+GBM.genes.of.interest2 <- rbind(H3K27M, WT)
+
+t_results <- list()
+for (i in genes_of_interest){
+  gene_temp <- GBM.genes.of.interest2[GBM.genes.of.interest2$gene_symbol == i,]
+  t_results[[i]] <- t.test(gene_temp$Expression27[gene_temp$type == "H3K27M"],gene_temp$Expression27[gene_temp$type == "WT"])
+}
+
+
+# Make a table to store the results
+t_results_sum <- data.frame(genes_of_interest,sapply(t_results, function(l) l[["statistic"]]), sapply(t_results, function(l) l[["p.value"]]))
```

Comment:
this seems to be a very neat way to present the t test results, which I didnt figure out in mine. Can you elaborate a bit more about this line of code, especially for this part that I dont quite understand: "...function(l) l[["statistic"]]), sapply(t_results, function(l) l[["p.value"]]))".